src/families/bitcoin/wallet-btc/crypto/bitcoincash.ts

```diff
@@ -7,16 +7,16 @@ import bchaddr from "bchaddrjs";
 import { toOutputScript } from "bitcoinjs-lib/src/address";
 import { InvalidAddress } from "@ledgerhq/errors";
 import { DerivationModes } from "../types";
-import { ICrypto } from "./types";
 import Base from "./base";
 
 // a mock explorer class that just use js objects
-class BitcoinCash implements ICrypto {
+class BitcoinCash extends Base {
```

Comment:
Did you check that all the default methods implemented in `Base` are relevant for all the coins you modified?

Reply:
good point.
I just changed the "Base" class.
In "Base" class, it has following public or protected methods: "constructor", "getPubkeyAt", "getLegacyAddress", "getAddress", "customGetAddress", "getDerivationMode", "toOutputScript", "validateAddress", "isTaprootAddress".
They are relevant for all the coins that inherited "Base" class, IMO

Two private methods in "Base": "getSegWitAddress", "getNativeSegWitAddress". They are not relevant for all the coins. I think it should be OK as they are private.